assembly/src/parsers/ast/ast.rs

```diff
@@ -0,0 +1,257 @@
+    U32CheckedMulImm(u32),
+    U32WrappingMul,
+    U32WrappingMulImm(u32),
+    U32OverflowingMul,
+    U32OverflowingMulImm(u32),
+    U32OverflowingMadd,
+    U32WrappingMadd,
+    U32CheckedDiv,
+    U32CheckedDivImm(u32),
+    U32UncheckedDiv,
+    U32UncheckedDivImm(u32),
+    U32CheckedMod,
+    U32CheckedModImm(u32),
+    U32CheckedDivMod,
+    U32CheckedDivModImm(u32),
+    U32UncheckedDivMod,
+    U32UncheckedDivModImm(u32),
+    U32CheckedAnd,
+    U32CheckedOr,
+    U32CheckedXor,
+    U32CheckedNot,
+    U32CheckedShr,
+    U32CheckedShrImm(u32),
+    U32UncheckedShr,
+    U32UncheckedShrImm(u32),
+    U32CheckedShl,
+    U32CheckedShlImm(u32),
+    U32UncheckedShl,
+    U32UncheckedShlImm(u32),
+    U32CheckedRotr,
+    U32CheckedRotrImm(u32),
+    U32UncheckedRotr,
+    U32UncheckedRotrImm(u32),
+    U32CheckedRotl,
+    U32CheckedRotlImm(u32),
+    U32UncheckedRotl,
+    U32UncheckedRotlImm(u32),
+    U32CheckedEq,
+    U32CheckedEqImm(u32),
+    U32CheckedNeq,
+    U32CheckedNeqImm(u32),
+    U32CheckedLt,
+    U32UncheckedLt,
+    U32CheckedLte,
+    U32UncheckedLte,
+    U32CheckedGt,
+    U32UncheckedGt,
+    U32CheckedGte,
+    U32UncheckedGte,
+    U32CheckedMin,
+    U32UncheckedMin,
+    U32CheckedMax,
+    U32UncheckedMax,
+
+    // ----- stack manipulation ---------------------------------------------------------------
+    Drop,
+    DropW,
+    PadW,
+    Dup0,
+    Dup1,
+    Dup2,
+    Dup3,
+    Dup4,
+    Dup5,
+    Dup6,
+    Dup7,
+    Dup8,
+    Dup9,
+    Dup10,
+    Dup11,
+    Dup12,
+    Dup13,
+    Dup14,
+    Dup15,
+    DupW0,
+    DupW1,
+    DupW2,
+    DupW3,
+    Swap,
+    Swap2,
+    Swap3,
+    Swap4,
+    Swap5,
+    Swap6,
+    Swap7,
+    Swap8,
+    Swap9,
+    Swap10,
+    Swap11,
+    Swap12,
+    Swap13,
+    Swap14,
+    Swap15,
+    SwapW,
+    SwapW1,
+    SwapW2,
+    SwapW3,
+    SwapDW,
+    MovUp2,
+    MovUp3,
+    MovUp4,
+    MovUp5,
+    MovUp6,
+    MovUp7,
+    MovUp8,
+    MovUp9,
+    MovUp10,
+    MovUp11,
+    MovUp12,
+    MovUp13,
+    MovUp14,
+    MovUp15,
+    MovUpW2,
+    MovUpW3,
+    MovDn2,
+    MovDn3,
+    MovDn4,
+    MovDn5,
+    MovDn6,
+    MovDn7,
+    MovDn8,
+    MovDn9,
+    MovDn10,
+    MovDn11,
+    MovDn12,
+    MovDn13,
+    MovDn14,
+    MovDn15,
+    MovDnW2,
+    MovDnW3,
+    CSwap,
+    CSwapW,
+    CDrop,
+    CDropW,
+
+    // ----- input / output operations --------------------------------------------------------
+    PushAdv(Felt),
+    PushEnvLocaddr(Felt),
+    PushEnvSdepth,
+    PushLocal(Felt),
+    PushMem,
+    PushMemImm(Felt),
+    PushWLocalImm(Felt),
+    PushWMem,
+    PushWMemImm(Felt),
+    PushConstants(Vec<Felt>),
+    PopLocalImm(Felt),
+    PopMem,
+    PopMemImm(Felt),
+    PopWMem,
+    PopWMemImm(Felt),
+    PopWLocalImm(Felt),
+    LoadWAdv,
+    LoadWMem,
+    LoadWMemImm(Felt),
+    LoadWLocalImm(Felt),
+    StoreWMem,
+    StoreWMemImm(Felt),
+    StoreMemImm(Felt),
+    StoreWLocalImm(u64),
+    AdvU64Div,
```

Comment:
The refactoring I mentioned has been merged earlier today. So, this will need to be updated to be consistent with [io ops](https://maticnetwork.github.io/miden/user_docs/assembly/io_operations.html).